classes/userGroup/Repository.php

```diff
@@ -350,6 +350,24 @@ public function getUserCountByContextId(?int $contextId = null): Collection
         return $this->dao->getUserCountByContextId($contextId);
     }
 
+    /**
+     * Get the user group a new author may be assigned to
+     * if they don't already have a role that can submit
+     *
+     * This returns the first user group with ROLE_ID_AUTHOR
+     * that permits self-registration.
+     */
+    public function getDefaultAuthorUserGroup(int $contextId): ?UserGroup
```

Comment:
The function name suggests that it'll *always* get the default user group, but this function won't behave that way. Do you think `getBestAuthorUserGroup`, `getCandidateAuthorUserGroup`, or `getPreferredAuthorUserGroup` would make more sense?

Reply:
I see what you mean. Since this is only used when determining what a user should submit as, what about `getFirstSubmitAsAuthorUserGroup()`? It's a mouthful, but that will prevent people from using it for other purposes, which is I think the confusion we want to avoid.

Reply:
...yes, it is a mouthful, but it's clear. Go for it.